src/org/ods/util/HelmStatusSimpleData.groovy

```diff
@@ -0,0 +1,186 @@
+                description = infoObject.description
+                lastDeployed = infoObject["last_deployed"]
+            }
+            hs.resourcesByKind = resourcesByKind.collectEntries { kind, names ->
+                [ kind, names.collect() ]
+            } as Map<String, List<String> >
+            return hs
+        } catch (IllegalArgumentException ex) {
+            throw new IllegalArgumentException(
+                "Unexpected helm status information in JSON at 'info': ${ex.message}")
+        }
+    }
+
+    @NonCPS
+    Map<String, List<String>> getResourcesByKind(List<String> kinds) {
+        return resourcesByKind.subMap(kinds)
+    }
+
+    @NonCPS
+    Map<String, Object> toMap() {
+        def result = [
+            name: name,
+            version: version,
+            namespace: namespace,
+            status: status,
+            description: description,
+            lastDeployed: lastDeployed,
+            resourcesByKind: resourcesByKind,
+        ]
+        return result
+    }
+
+    @NonCPS
+    String toString() {
+        return toMap().toMapString()
+    }
+
+    @NonCPS
+    private static Tuple2<String, String> extractResource(
+        resourceJsonObject, String context) {
+        def resourceObject = ensureMap(resourceJsonObject, context)
+        Map<String, Object> resource = [:]
+        if (resourceObject?.kind) {
+            resource.kind = resourceObject.kind
+        }
+        if (resource?.kind == "PodList") {
+            return null
+        }
+        Map<String, Object> metadataObject = ensureMap(
+            resourceObject.metadata, "${context}.metadata")
+        if (metadataObject?.name) {
+            resource["name"] = metadataObject.name
+        }
+
+        def expected = collectMissingStringAttributes(resource, ["name", "kind"])
+        handleMissingKeysOrBadTypes(expected.first, expected.second)
+        return new Tuple2(resource.kind, resource.name)
+    }
+
+    @SuppressWarnings(['Instanceof'])
+    @NonCPS
+    private static Map ensureMap(Object obj, String context) {
+        if (obj == null) {
+            return [:]
+        }
+        if (!(obj instanceof Map)) {
+            def msg = context ?
+                "${context}: expected JSON object, found ${obj.getClass()}" :
+                "Expected JSON object, found ${obj.getClass()}"
+
+            throw new IllegalArgumentException(msg)
+        }
+        return obj as Map
+    }
+
+    @SuppressWarnings(['Instanceof'])
+    @NonCPS
+    private static List ensureList(Object obj, String context) {
+        if (obj == null) {
+            return []
+        }
+        if (!(obj instanceof List)) {
+            throw new IllegalArgumentException(
+                "${context}: expected JSON array, found ${obj.getClass()}")
+        }
+        return obj as List
+    }
+
+    @SuppressWarnings(['Instanceof'])
+    @NonCPS
+    private static Tuple2<List<String>, List<String>> collectMissingStringAttributes(
```

Comment:
If an attribute is missing, its value will be null. The users of this class will decide what to do with it.

Reply:
Same as: https://github.com/opendevstack/ods-jenkins-shared-library/pull/1143#discussion_r1806269048